woocommerce/woocommerce
# Improved shipping calculator experience

### All Submissions:






### Changes proposed in this Pull Request:

- Allows the shipping calculator toggle/button to be customised
- Hides rates when there are none, and the shipping calculator is disabled
- Shows added context when showing rates; either making clear it's an estimate, informing the user they need to checkout, or showing them their current destination.

Screenshots:

![monosnap 2018-06-01 21-35-38](https://user-images.githubusercontent.com/90977/40862229-c57bd114-65e3-11e8-9f74-3b3458febaca.png)
![monosnap 2018-06-01 21-34-01](https://user-images.githubusercontent.com/90977/40862231-c6fa64d8-65e3-11e8-99a5-fd9926623cc4.png)
![monosnap 2018-06-01 21-34-32](https://user-images.githubusercontent.com/90977/40862233-c8a95410-65e3-11e8-9cd2-85a08892216f.png)
![monosnap 2018-06-01 21-34-54](https://user-images.githubusercontent.com/90977/40862236-ca5b197e-65e3-11e8-92c6-0d657e14b354.png)

Closes #19686

### How to test the changes in this Pull Request:

1. In all supported themes...
2. Go to cart in incognito browser. See results.
3. Calculate shipping. Check location is shown.
4. Location should not be shown on checkout.

### Changelog entry

> Improved shipping method template and shipping calculator to include the location being estimated, and to improve the wording of the shipping calculator button based on context.
